packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -852,6 +860,12 @@ class _ToggleButton extends StatelessWidget {
   /// The button's label, which is usually an [Icon] or a [Text] widget.
   final Widget child;
 
+  Color _resolveColor(Set<MaterialState> states, Color? widgetColor, Color? themeColor, Color defaultColor) {
```

Comment:
This doesn't look right. Here's what I would expect:

```dart
@immutable
class _DefaultFillColor extends MaterialStateProperty<Color> {
  _DefaultFillColor(this.colorScheme);

  final ColorScheme colorScheme;

  @override
  Color resolve(Set<MaterialState> states) {
    if(states.contains(MaterialState.selected)) {
      return colorScheme.primary.withOpacity(0.12);
    }
    return colorScheme.surface.withOpacity(0.0);
  }
}

Color _resolveColor(Set<MaterialState> states, Color? widgetColor, Color? themeColor, MaterialStateProperty<Color> defaultColor) {
  return MaterialStateProperty.resolveAs<Color?>(widgetColor, states)
      ?? MaterialStateProperty.resolveAs<Color?>(themeColor, states)
      ?? defaultColor.resolve(states);
}
```

And then down below:
```dart
  currentFillColor = _resolveColor(states, fillColor, toggleButtonsTheme.fillColor, _DefaultFillColor(theme.colorScheme));
```

Reply:
I have added the `_DefaultFillColor` class and made some change to `_resolveColor`
```dart
  Color _resolveColor(Set<MaterialState> states, MaterialStateProperty<Color?> widgetColor,
  MaterialStateProperty<Color?> themeColor, MaterialStateProperty<Color> defaultColor) {
  return widgetColor.resolve(states)
    ?? themeColor.resolve(states)
    ?? defaultColor.resolve(states);
  }
```
and the `currentFillColor` to
```
     final Color currentFillColor = _resolveColor(states, _ResolveFillColor(fillColor),
    _ResolveFillColor(toggleButtonsTheme.fillColor), _DefaultFillColor(theme.colorScheme));
```